### PR TITLE
Gitian based binary and docker build

### PR DESCRIPTION
**Pull request by alpe**
Build Tendermint via gitian on CI

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -15,33 +23,58 @@ go:
...
-  - GORACE="halt_on_error=1"
-    SOURCE_VERSION=$(source ./build-version.txt && echo -n $SOURCE_VERSION)
-    SOURCE_REPO=$(source ./build-version.txt && echo -n $SOURCE_REPO)
-    BUILD_VERSION=$(echo ${TRAVIS_COMMIT} | cut -c 1-10)
+  global:
+    - GORACE="halt_on_error=1"
+    - BUILD_TAG=$(source ./build-version.txt && echo -n $BUILD_TAG)
+    - BUILD_COMMIT=$(source ./build-version.txt && echo -n $BUILD_COMMIT)
+    - IMAGE_NAME="iov1/tendermint:${BUILD_TAG}"
+    - BUILD_VERSION=$(echo ${TRAVIS_COMMIT} | cut -c 1-10) # random id
```
Comment (husio) [inline .travis.yml]: This is not a random ID. Is this on purpose?

File: README.md [hunk 2]
```diff
@@ -1,15 +1,11 @@
...
-```
-
-* ` --no-cache` to not cache the code during checkout step
-
+## Tendermint-build
+[Gitian](https://gitian.org) base build process for [Tendermint](https://github.com/tendermint/tendermint).
+
+#### Trigger new artifact build
+1. Update `./build-version.txt` file and set proper tag and commit hash.
+2. Do PR and let travisCI build it.
```
Comment (husio) [inline README.md]: I think it would be better to write

> Create a Pull Request and let Travis CI build it.

:nail_care:

File: README.md [hunk 3]
```diff
@@ -1,15 +1,11 @@
...
+3. Merge to master and Tag with `v0.31.5-iov1`. The first part is the Tendermint tag you are building and the second a iov counter that you can increase if you have to.
```
Comment (husio) [inline README.md]: I understand that `v0.31.5-iov1` is a string build using the version information. Maybe it is better to write it

> Merge the Pull Request to `master` branch and create a tag `BUILD_TAG-iovCOUNTER`, (for example `v0.31.5-iov1`)
>  `BUILD_TAG` is the Tendermint tag you are building and the `COUNTER` an IOV counter that you can increase if you have to, to create another build for the same build tag value.

File: .travis.yml (hunk 1 above)
Comment (alpe) [reply .travis.yml]: ja, any unique id is good enough...

**Pull request merged**